chains/manager.go

```diff
@@ -736,25 +733,29 @@ func (m *manager) createSnowmanChain(
 		return nil, fmt.Errorf("couldn't initialize sender: %w", err)
 	}
 
-	// first vm to be init is P-Chain once, which provides validator interface to all ProposerVMs
-	if m.validatorState == nil {
-		if m.ManagerConfig.StakingEnabled {
-			valState, ok := vm.(validators.State)
-			if !ok {
-				return nil, fmt.Errorf("expected validators.State but got %T", vm)
-			}
-
-			// Initialize the validator state for future chains.
-			m.validatorState = validators.NewLockedState(&ctx.Lock, valState)
-
-			// Notice that this context is left unlocked. This is because the
-			// lock will already be held when accessing these values on the
-			// P-chain.
-			ctx.ValidatorState = valState
-		} else {
-			m.validatorState = validators.NewNoState()
-			ctx.ValidatorState = m.validatorState
+	// If the platform VM state is `nil`, we are initializing the first chain,
+	// which is the platform VM, so we should set this field to the type-asserted
+	// platform VM handle.
+	if m.platformVMState == nil {
+		platformVMState, ok := vm.(platform.VMState)
+		if !ok {
+			return nil, fmt.Errorf("first VM should implement `platform.VMState`")
 		}
+		m.platformVMState = platformVMState
+		ctx.PlatformVMState = platformVMState
```

Comment:
You are reading wrong. It only set the validator state to `NewNoState()` previously when staking was disabled. Staking is now determined at the EVM level, so that has setting has no bearing on our staking anymore.

You also seem to misunderstand all of the changes in this file, even though I believed I already addressed it in a previous comment?

So again:
- the platform VM was previously responsible for both providing the height of the platform chain, and for providing a list of validators per chain;
- we changed that to providing one list of validators for every chain, and this PR takes it a step further by only keeping one validator set and removing the validator manager that is not needed;
- we thus remove the validator management related function from the platform chain, and use `platform.VMState` as an interface to describe a component that can return the platform chain height;
- we then use a new separate interface for retrieving validators from a VM, so we can handle those two roles separately, one by the platform VM, one by the EVM.